source/common/upstream/outlier_detection_impl.cc

```diff
@@ -186,15 +208,34 @@ bool DetectorImpl::enforceEjection(EjectionType type) {
   NOT_REACHED;
 }
 
+void DetectorImpl::updateEnforcedEjectionStats(EjectionType type) {
+  stats_.ejections_enforced_total_.inc();
+  switch (type) {
+  case EjectionType::SuccessRate:
+    stats_.ejections_enforced_success_rate_.inc();
+    break;
+  case EjectionType::Consecutive5xx:
+    stats_.ejections_enforced_consecutive_5xx_.inc();
+    break;
+  case EjectionType::ConsecutiveGatewayFailure:
+    stats_.ejections_enforced_consecutive_gateway_failure_.inc();
+    break;
+  }
+}
+
 void DetectorImpl::ejectHost(HostSharedPtr host, EjectionType type) {
   uint64_t max_ejection_percent = std::min<uint64_t>(
       100, runtime_.snapshot().getInteger("outlier_detection.max_ejection_percent",
                                           config_.maxEjectionPercent()));
   double ejected_percent = 100.0 * stats_.ejections_active_.value() / host_monitors_.size();
   if (ejected_percent < max_ejection_percent) {
-    stats_.ejections_total_.inc();
+    if (type == EjectionType::Consecutive5xx || type == EjectionType::SuccessRate) {
+      // Deprecated counter, preserving old behaviour until it's removed.
```

Comment:
Can you add this to https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md?

Reply:
Done in f782640